src/Umbraco.Web/Templates/TemplateUtilities.cs

```diff
@@ -45,18 +51,22 @@ public static string ParseInternalLinks(string text, UrlProvider urlProvider)
                     var id = tag.Groups[1].Value; //.Remove(tag.Groups[1].Value.Length - 1, 1);
 
                     //The id could be an int or a UDI
-                    Udi udi;
-                    if (Udi.TryParse(id, out udi))
+                    if (Udi.TryParse(id, out var udi))
                     {
                         var guidUdi = udi as GuidUdi;
                         if (guidUdi != null)
                         {
-                            var newLink = urlProvider.GetUrl(guidUdi.Guid);
+                            var newLink = "#";
+                            if (guidUdi.EntityType == Constants.UdiEntityType.Document)
+                                newLink = urlProvider.GetUrl(guidUdi.Guid);
+                            else if (guidUdi.EntityType == Constants.UdiEntityType.Media)
+                                newLink = mediaCache.GetById(guidUdi.Guid)?.Url;
```

Comment:
If this is null, do we want to still keep the `newLink` as "#" ?

Reply:
Not sure, I looked at the `UrlProvider` returns "#" if it cannot resolve the url, but maybe we should just set it to an empty value instead?

Reply:
I'm pretty sure that the "#" is used as a special case elsewhere, meaning i think that's the expected value in some places when a URL cannot be resolved so i think we need to keep it that way.

Reply:
Yep, the "#" from the `UrlProvider` is used somewhere, I was just thinking we could change the `newLink` to an empty string instead of "#", but not sure if it makes sence

Reply:
@rasmusjp  We are going to need to keep the result of "#", we can't just go changing how things work at this stage, there's definitely some things checking for the '#' value.

Reply:
@Shazwazza As mentioned below I'm a little confused about what you wan't me to change since I haven't changed anything in the `UrlProvider`, it still returns '#' if the url can't be resolved.

Reply:
@rasmusjp no it doesn't, your code changes this to return null: `newLink = mediaCache.GetById(guidUdi.Guid)?.Url;` if the `GetById` call returns null

Reply:
@Shazwazza ahh okay, I misunderstood. I thought you were talking about the actual return value from `urlProvider.GetUrl(guidUdi.Guid)` (since the '#' from that is used elsewhere) and not what the `href` was being set to. I've updated the code so `href` is being set '#' if `newLink` is null.